stream_alert/shared/lookup_tables/driver_s3.py

```diff
@@ -120,6 +144,131 @@ def _reload(self):
         the file's contents into memory.
         """
         # First, download the item from S3
+        data = self._s3_adapter.download()
+
+        # The lookup data can optionally be compressed, so try to decompress
+        # This will fall back and use the original data if decompression fails
+        if self._compression:
+            data = Compression.gz_decompress(self, data)
+        else:
+            LOGGER.debug('LookupTable (%s): File does not need decompression')
+
+        # Decode the data; right now we make the assumption that the data is always encoded
+        # as JSON.
+        data = Encoding.json_decode(self, data)
+
+        self._cache.setall(data, self._cache_refresh_minutes)
+        self._cache.set(self._MAGIC_CACHE_TTL_KEY, True, self._cache_refresh_minutes)
+        LOGGER.info('LookupTable (%s): Successfully loaded', self.id)
+
+
+class Compression:
```

Comment:
up to you if you want to take this suggestion - but you could have the `S3Driver` (and other drivers that might use encoding/compression) use multi-inheritance for this, like:

```
class S3Driver(Compression, Encoding, PersistenceDriver):
    ...
```

so these static methods would be accessible from within the class itself. then again, it might be hard with the `id` stuff.. aside from passing in `self` as an arg

Reply:
EDIT: actually I lied.. using a mix-in like that, you would have to convert the methods from `staticmethod`s to instance methods, and `self.id` would have the `id` attribute accessible from `S3Driver`.. again your call! but I know you like mixins :P

Reply:
also - could even use `S3Adapter` as a mixin along with these? unsure, since after further review it sorta seems like you're moving _away_ from this approach?

Reply:
There was no good reason I used static methods. My idea with the `Compression` class was to have another layer where we could support multiple types of compression (gzip being one), that the user could configure. In this way a mixin wouldn't work and we'd have to resort to composition... but static methods don't make this obvious